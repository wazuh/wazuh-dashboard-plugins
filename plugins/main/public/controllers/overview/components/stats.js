/*
 * Wazuh app - React component for showing stats about agents.
 *
 * Copyright (C) 2015-2022 Wazuh, Inc.
 *
 * This program is free software; you can redistribute it and/or modify
 * it under the terms of the GNU General Public License as published by
 * the Free Software Foundation; either version 2 of the License, or
 * (at your option) any later version.
 *
 * Find more information about this on the LICENSE file.
 */
import React, { Component } from 'react';
import PropTypes from 'prop-types';
import {
  EuiStat,
  EuiFlexItem,
  EuiFlexGroup,
  EuiPage,
  EuiToolTip,
} from '@elastic/eui';
import { withErrorBoundary } from '../../../components/common/hocs';
import { API_NAME_AGENT_STATUS } from '../../../../common/constants';
import {
  agentStatusLabelByAgentStatus,
  agentStatusColorByAgentStatus,
} from '../../../../common/services/wz_agent_status';
import { getCore } from '../../../kibana-services';
import { endpointSummary } from '../../../utils/applications';
import { LastAlertsStat } from './last-alerts-stat';

export const Stats = withErrorBoundary(
  class Stats extends Component {
    constructor(props) {
      super(props);

      this.state = {};
      this.agentStatus = [
        API_NAME_AGENT_STATUS.ACTIVE,
        API_NAME_AGENT_STATUS.DISCONNECTED,
      ].map(status => ({
        status,
        label: agentStatusLabelByAgentStatus(status),
        onClick: () => this.goToAgents(status),
        color: agentStatusColorByAgentStatus(status),
      }));
    }

    goToAgents(status) {
      if (status) {
        sessionStorage.setItem(
          'wz-agents-overview-table-filter',
          JSON.stringify({ q: `id!=000;status=${status}` }),
        );
      } else if (sessionStorage.getItem('wz-agents-overview-table-filter')) {
        sessionStorage.removeItem('wz-agents-overview-table-filter');
      }
      getCore().application.navigateToApp(endpointSummary.id, {
        path: '#/agents-preview',
      });
    }

    renderTitle(total) {
      return (
        <EuiToolTip position='top' content={`Go to all agents`}>
          <span>{total}</span>
        </EuiToolTip>
      );
    }

    render() {
      return (
        <EuiPage>
          <EuiFlexGroup>
            <EuiFlexItem />
            {this.agentStatus.map(({ status, label, onClick, color }) => (
              <EuiFlexItem key={`agent-status-${status}`}>
                <EuiStat
                  title={
                    <EuiToolTip
                      position='top'
                      content={`Go to ${label.toLowerCase()} agents`}
                    >
                      <span
                        className='statWithLink'
                        style={{ cursor: 'pointer' }}
                        onClick={onClick}
                      >
                        {typeof this.props[status] !== 'undefined'
                          ? this.props[status]
                          : '-'}
                      </span>
                    </EuiToolTip>
                  }
                  description={`${label} agents`}
                  titleColor={color}
                  textAlign='center'
                />
              </EuiFlexItem>
            ))}
            <LastAlertsStat />
            <EuiFlexItem />
          </EuiFlexGroup>
        </EuiPage>
      );
    }
  },
);

Stats.propTypes = {
  active: PropTypes.any,
  disconnected: PropTypes.any,
};
