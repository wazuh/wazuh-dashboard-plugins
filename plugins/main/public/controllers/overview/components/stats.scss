.vulnerabilites-summary-card {
  padding-top: 2vh;
}
