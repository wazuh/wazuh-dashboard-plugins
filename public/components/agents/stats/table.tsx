/*
 * Wazuh app - Component to display the table in the Agent stats
 * Copyright (C) 2015-2022 Wazuh, Inc.
 *
 * This program is free software; you can redistribute it and/or modify
 * it under the terms of the GNU General Public License as published by
 * the Free Software Foundation; either version 2 of the License, or
 * (at your option) any later version.
 *
 * Find more information about this on the LICENSE file.
 */
import React from 'react';
import {
  EuiButtonEmpty,
  EuiFlexGroup,
  EuiFlexItem,
  EuiHorizontalRule,
  EuiIcon,
  EuiInMemoryTable,
  EuiLoadingSpinner,
  EuiPanel,
  EuiSpacer,
  EuiText,
} from '@elastic/eui';
import * as FileSaver from '../../../services/file-saver';
import { formatUIDate } from '../../../react-services';
import {
  UI_ERROR_SEVERITIES,
  UIErrorLog,
  UIErrorSeverity,
  UILogLevel,
} from '../../../react-services/error-orchestrator/types';
import { UI_LOGGER_LEVELS } from '../../../../common/constants';
import { getErrorOrchestrator } from '../../../react-services/common-services';

export function AgentStatTable({ columns, title, start, end, loading, items, exportCSVFilename }) {
  return (
    <EuiPanel>
      <EuiFlexGroup justifyContent="spaceBetween">
        <EuiFlexItem grow={false}>
          <EuiText>{title}</EuiText>
        </EuiFlexItem>
        <EuiFlexItem grow={false}>
          <EuiText>
            <EuiIcon type="calendar" /> Start:{' '}
            {loading ? <EuiLoadingSpinner size="s" /> : start ? formatUIDate(start) : '-'} - End:{' '}
            {loading ? <EuiLoadingSpinner size="s" /> : end ? formatUIDate(end) : '-'}
          </EuiText>
        </EuiFlexItem>
      </EuiFlexGroup>
      <EuiHorizontalRule margin="xs" />
      <EuiInMemoryTable columns={columns} items={items || []} loading={loading} pagination={true} />
      <EuiSpacer size="xs" />
      <EuiFlexGroup justifyContent="flexEnd">
        <EuiFlexItem grow={false}>
          <EuiButtonEmpty
            onClick={() => downloadCsv(columns, items, exportCSVFilename)}
            iconType="importAction"
            isDisabled={loading}
          >
            Download CSV
          </EuiButtonEmpty>
        </EuiFlexItem>
      </EuiFlexGroup>
    </EuiPanel>
  );
}

function downloadCsv(columns: any[], data: any[], filename: string) {
  try {
    const header = columns.map((column) => column.name).join(',');
    const body = data.map((row) => columns.map((column) => row[column.field]).join(',')).join('\n');
    const result = [header, body].join('\n');
    const blob = new Blob([result], { type: 'text/csv' }); // eslint-disable-line
    FileSaver.saveAs(blob, `${filename}.csv`);
  } catch (error) {
    const options: UIErrorLog = {
      context: 'downloadCsv',
      level: UI_LOGGER_LEVELS.ERROR as UILogLevel,
      severity: UI_ERROR_SEVERITIES.BUSINESS as UIErrorSeverity,
      error: {
        error: error,
        message: `Error generating CSV: ${error.message}`,
        title: `CSV: ${error.name}`,
      },
    };
    getErrorOrchestrator().handleError(options);
  }
}
