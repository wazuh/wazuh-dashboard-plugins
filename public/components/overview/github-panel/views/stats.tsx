/*
 * Wazuh app - GitHub Panel tab - Stats
 * Copyright (C) 2015-2022 Wazuh, Inc.
 *
 * This program is free software; you can redistribute it and/or modify
 * it under the terms of the GNU General Public License as published by
 * the Free Software Foundation; either version 2 of the License, or
 * (at your option) any later version.
 *
 * Find more information about this on the LICENSE file.
 */


import React from 'react';
import { EuiDescriptionList, EuiPanel } from '@elastic/eui';
import { PanelModuleConfiguration } from '../../../common/modules/panel';
import { renderValueNoThenEnabled } from '../../../../controllers/management/components/management/configuration/utils/utils';
import { LogoGitHub } from '../../../common/logos';

const settings = [
  { field: 'enabled', label: 'Service status', render: renderValueNoThenEnabled },
  { field: 'only_future_events', label: 'Collect events generated since Wazuh agent was started'},
  { field: 'time_delay', label: 'Time in seconds that each scan will monitor until that delay backwards'},
  { field: 'curl_max_size', label: 'Maximum size allowed for the GitHub API response'},
  { field: 'interval', label: 'Interval between GitHub wodle executions in seconds'},
  { field: 'event_type', label: 'Event type'},
  { field: 'api_auth', label: 'Credentials', render: (value) => value.map(v => 
    <EuiPanel paddingSize='s' key={`module_configuration_api_auth_${v.org_name}_${v.client_id}`}>
      <EuiDescriptionList listItems={[
        {title: 'Organization', description: v.org_name},
        {title: 'Token', description: v.api_token}
      ].filter(item => typeof item.description !== 'undefined')}/>
    </EuiPanel>
  ).reduce((prev, cur) => [prev, <div key={`padding-len-${prev.length}`} style={{marginTop: '8px'}} /> , cur], [])}
];

const mapWModuleConfigurationToRenderProperties = (wmodules: {[key: string]: any}[], wmoduleID: string, entity: string, name: string = '') => {
  const configuration = wmodules.find(wmodule => Object.keys(wmodule)[0] === wmoduleID);
  return configuration 
    ? {entity, name, configuration: configuration[Object.keys(configuration)[0]]}
    : null;
};

export const ModuleConfiguration = props => <PanelModuleConfiguration 
  moduleTitle='GitHub'
  moduleIconType={LogoGitHub}
  settings={settings}
  configurationAPIPartialPath='/wmodules/wmodules'
  mapResponseConfiguration={(response, type, params) => {
    return type === 'agent'
      ? mapWModuleConfigurationToRenderProperties(response.data.data.wmodules, 'github', 'Agent', params.name)
      : (type === 'cluster_node'
      ? mapWModuleConfigurationToRenderProperties(response.data.data.affected_items[0].wmodules, 'github', 'Manager', params.name)
      : mapWModuleConfigurationToRenderProperties(response.data.data.affected_items[0].wmodules, 'github', 'Manager')
    );
  }}
/>
