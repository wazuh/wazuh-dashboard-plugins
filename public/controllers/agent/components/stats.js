/*
 * Wazuh app - React component for showing agent fields such as IP, ID, name,
 * version, OS, registration date, last keep alive.
 *
 * Copyright (C) 2015-2020 Wazuh, Inc.
 *
 * This program is free software; you can redistribute it and/or modify
 * it under the terms of the GNU General Public License as published by
 * the Free Software Foundation; either version 2 of the License, or
 * (at your option) any later version.
 *
 * Find more information about this on the LICENSE file.
 */
import React, { Component } from 'react';
import PropTypes from 'prop-types';
import { EuiStat, EuiFlexItem, EuiFlexGroup, EuiPanel } from '@elastic/eui';

export class Stats extends Component {
  constructor(props) {
    super(props);

    this.state = {};
  }

  buildStats(items) {
    const stats = items.map(item => {
      return (
        <EuiFlexItem key={item.description} style={item.style || null}>
          <EuiStat
            title={item.title}
            description={item.description}
            textAlign="center"
            titleSize="s"
            reverse
          />
        </EuiFlexItem>
      );
    });
    return stats;
  }

  render() {
    const stats = this.buildStats([
      { title: this.props.id, description: 'ID', style: { maxWidth: 100 } },
      { title: this.props.ip, description: 'IP' },
      { title: this.props.version, description: 'Version' },
      {
        title: this.props.agentOS,
        description: 'OS',
        style: { minWidth: 400 }
      },
      { title: this.props.dateAdd, description: 'Registration date' },
      { title: this.props.lastKeepAlive, description: 'Last keep alive' }
    ]);
    return (
      <EuiPanel betaBadgeLabel={this.props.name}>
        <EuiFlexGroup>{stats}</EuiFlexGroup>
      </EuiPanel>
    );
  }
}

Stats.propTypes = {
  id: PropTypes.string,
  ip: PropTypes.string,
  version: PropTypes.string,
  agentOS: PropTypes.string,
  dateAdd: PropTypes.any,
  lastKeepAlive: PropTypes.any,
  name: PropTypes.string
};
