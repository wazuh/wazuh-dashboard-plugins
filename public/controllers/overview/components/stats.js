/*
 * Wazuh app - React component for showing stats about agents.
 *
 * Copyright (C) 2015-2022 Wazuh, Inc.
 *
 * This program is free software; you can redistribute it and/or modify
 * it under the terms of the GNU General Public License as published by
 * the Free Software Foundation; either version 2 of the License, or
 * (at your option) any later version.
 *
 * Find more information about this on the LICENSE file.
 */
import React, { Component } from 'react';
import PropTypes from 'prop-types';
import { EuiStat, EuiFlexItem, EuiFlexGroup, EuiPage, EuiToolTip } from '@elastic/eui';
import { withErrorBoundary } from '../../../components/common/hocs';
import { UI_ORDER_AGENT_STATUS } from '../../../../common/constants';
import { agentStatusLabelByAgentStatus, agentStatusColorByAgentStatus } from '../../../../common/services/wz_agent_status';

export const Stats = withErrorBoundary (class Stats extends Component {
  constructor(props) {
    super(props);

    this.state = {};
    this.agentStatus = ['total', ...UI_ORDER_AGENT_STATUS].map(status => ({
      status,
      label: status !== 'total' ? agentStatusLabelByAgentStatus(status) : 'Total',
      onClick: () => this.goToAgents(status !== 'total' ? status : null),
      color: status !== 'total' ? agentStatusColorByAgentStatus(status) : 'primary'
    }));
  }

  goToAgents(status) {
    if(status){
      sessionStorage.setItem(
        'agents_preview_selected_options',
        JSON.stringify([{field: 'q', value: `status=${status}`}])
      );
    }else if(sessionStorage.getItem('agents_preview_selected_options')){
      sessionStorage.removeItem('agents_preview_selected_options');
    }
    window.location.href = '#/agents-preview';
  }

  renderTitle(total) {
    return <EuiToolTip position="top" content={`Go to all agents`}>
      <span>
        {total}
      </span>
    </EuiToolTip>
  }

  render() {
    return (
      <EuiPage>
        <EuiFlexGroup>
          <EuiFlexItem />
            {this.agentStatus.map(({status, label, onClick, color}) => (
              <EuiFlexItem key={`agent-status-${status}`}>
                <EuiStat
                  title={
                    <EuiToolTip position="top" content={`Go to ${label.toLowerCase()} agents`}>
                      <span
                        className='statWithLink'
                        style={{ cursor: "pointer" }}
                        onClick={onClick}
                      >
                        {typeof this.props[status] !== 'undefined' ? this.props[status] : '-'}
                      </span>
                    </EuiToolTip>
                  }
                  description={`${label} agents`}
                  titleColor={color}
                  textAlign="center"
                />
              </EuiFlexItem>
            ))}
          <EuiFlexItem />
        </EuiFlexGroup>
      </EuiPage>
    );
  }
});

Stats.propTypes = {
  total: PropTypes.any,
  active: PropTypes.any,
  disconnected: PropTypes.any,
  pending: PropTypes.any,
  never_connected: PropTypes.any
};
