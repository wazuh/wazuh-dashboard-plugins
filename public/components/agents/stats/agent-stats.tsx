/*
 * Wazuh app - Component to display the Agent stats
 * Copyright (C) 2015-2021 Wazuh, Inc.
 *
 * This program is free software; you can redistribute it and/or modify
 * it under the terms of the GNU General Public License as published by
 * the Free Software Foundation; either version 2 of the License, or
 * (at your option) any later version.
 *
 * Find more information about this on the LICENSE file.
 */
import React, { useState, useEffect } from 'react';
import { 
  EuiFlexGroup,
  EuiFlexItem,
  EuiLoadingSpinner,
  EuiPanel,
  EuiPage,
  EuiPageBody,
  EuiSpacer,
  EuiText
} from '@elastic/eui';

import { withGlobalBreadcrumb, withReduxProvider, withGuard, withUserAuthorizationPrompt } from '../../common/hocs';
import { compose } from 'redux';
import { WzRequest, formatUIDate } from '../../../react-services';
import { AgentStatTable } from './table';
import { PromptNoActiveAgentWithoutSelect, PromptAgentFeatureVersion } from '../prompts';

const tableColumns = [
  {
    field: 'location',
    name: 'Location',
    sortable: true,
  },
  {
    field: 'events',
    name: 'Events',
    sortable: true,
  },
  {
    field: 'bytes',
    name: 'Bytes',
    sortable: true,
  }
];

const statsAgents: {title: string, field: string, render?: (value) => any}[] = [
  {
    title: 'Status',
    field: 'status',
  },
  {
    title: 'Buffer',
    field: 'buffer_enabled',
    render: (value) => value ? 'enabled' : 'disabled'
  },
  {
    title: 'Message buffer',
    field: 'msg_buffer'
  },
  {
    title: 'Messages count',
    field: 'msg_count'
  },
  {
    title: 'Messages sent',
    field: 'msg_sent'
  },
  {
    title: 'Last ack',
    field: 'last_ack',
    render: formatUIDate
  },
  {
    title: 'Last keep alive',
    field: 'last_keepalive',
    render: formatUIDate
  }
];

export const MainAgentStats = compose(
  withReduxProvider,
  withGlobalBreadcrumb(({agent}) => [
    {
      text: ''
    },
    {
      text: 'Agents',
      href: "#/agents-preview"
    },
    { agent },
    {
      text: 'Stats'
    },
  ]),
  withUserAuthorizationPrompt(({agent}) => [[
    {action: 'agent:read', resource: `agent:id:${agent.id}`}, 
    ...(agent.group || []).map(group => ({ action: 'agent:read', resource: `agent:group:${group}` }))
  ]]),
  withGuard(({agent}) => agent.status !== 'active', PromptNoActiveAgentWithoutSelect),
  withGuard(({agent}) => {
    const [major, minor, patch] = agent.version.replace('Wazuh v','').split('.').map(value => parseInt(value));
    return !(major >= 4 && minor >= 2 && patch >= 0)
  }, () => <PromptAgentFeatureVersion version='equal or higher version than 4.2.0'/>)
)(AgentStats);

function AgentStats({agent}){
  const [loading, setLoading] = useState();
  const [dataStatLogcollector, setDataStatLogcollector] = useState({});
  const [dataStatAgent, setDataStatAgent] = useState();
  useEffect(() => {
    (async function(){
      setLoading(true);
      try{
        const responseDataStatLogcollector = await WzRequest.apiReq('GET', `/agents/${agent.id}/stats/logcollector`, {});
        const responseDataStatAgent = await WzRequest.apiReq('GET', `/agents/${agent.id}/stats/agent`, {});
        setDataStatLogcollector(responseDataStatLogcollector?.data?.data?.affected_items?.[0] || {});
        setDataStatAgent(responseDataStatAgent?.data?.data?.affected_items?.[0] || undefined);
      }catch(error){

      }
      setLoading(false);
    })()
  }, []);
  return (
    <EuiPage>
      <EuiPageBody>
        <EuiFlexGroup>
          <EuiFlexItem>
            <EuiPanel paddingSize="m">
              <EuiFlexGroup>
                {statsAgents.map(stat => (
                  <EuiFlexItem key={`agent-stat-${stat.field}`} grow={false}>
                    <EuiText>{stat.title}: {loading ? <EuiLoadingSpinner size="s" /> : <strong>{dataStatAgent !== undefined ? (stat.render ? stat.render(dataStatAgent[stat.field]) : dataStatAgent?.[stat.field]) : '-'}</strong>}</EuiText>
                  </EuiFlexItem>
                ))}
              </EuiFlexGroup>
            </EuiPanel>
          </EuiFlexItem>
        </EuiFlexGroup>
        <EuiSpacer />
        <EuiFlexGroup>
          <EuiFlexItem>
            <AgentStatTable
              columns={tableColumns}
              loading={loading}
              title='Global'
              start={dataStatLogcollector?.global?.start}
              end={dataStatLogcollector?.global?.end}
              items={dataStatLogcollector?.global?.files}
              exportCSVFilename={`agent-stats-${agent.id}-logcollector-global`}
            />
          </EuiFlexItem>
          <EuiFlexItem>
            <AgentStatTable
              columns={tableColumns}
              loading={loading}
              title='Interval'
              start={dataStatLogcollector?.interval?.start}
              end={dataStatLogcollector?.interval?.end}
              items={dataStatLogcollector?.interval?.files}
              exportCSVFilename={`agent-stats-${agent.id}-logcollector-interval`}
            />
          </EuiFlexItem>
        </EuiFlexGroup>
      </EuiPageBody>
    </EuiPage>
  )
}
