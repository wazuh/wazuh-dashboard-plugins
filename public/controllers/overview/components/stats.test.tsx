/*
 * Wazuh app - React test for Stats component.
 *
 * Copyright (C) 2015-2022 Wazuh, Inc.
 *
 * This program is free software; you can redistribute it and/or modify
 * it under the terms of the GNU General Public License as published by
 * the Free Software Foundation; either version 2 of the License, or
 * (at your option) any later version.
 *
 * Find more information about this on the LICENSE file.
 *
 */

import React from 'react';
import { mount } from 'enzyme';
import { Stats } from './stats';

describe('Stats component', () => {
  it('renders correctly to match the snapshot', () => {
    const wrapper = mount(<Stats />);

    expect(wrapper).toMatchSnapshot();
  });
});
