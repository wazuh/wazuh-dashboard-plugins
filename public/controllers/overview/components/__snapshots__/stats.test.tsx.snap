// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Stats component renders correctly to match the snapshot 1`] = `
<Component>
  <ErrorBoundary>
    <withErrorBoundary(Stats)>
      <EuiPage>
        <div
          className="euiPage"
        >
          <EuiFlexGroup>
            <div
              className="euiFlexGroup euiFlexGroup--gutterLarge euiFlexGroup--directionRow euiFlexGroup--responsive"
            >
              <EuiFlexItem>
                <div
                  className="euiFlexItem"
                />
              </EuiFlexItem>
              <EuiFlexItem
                key="agent-status-total"
              >
                <div
                  className="euiFlexItem"
                >
                  <EuiStat
                    description="Total agents"
                    textAlign="center"
                    title={
                      <EuiToolTip
                        content="Go to total agents"
                        delay="regular"
                        position="top"
                      >
                        <span
                          className="statWithLink"
                          onClick={[Function]}
                          style={
                            Object {
                              "cursor": "pointer",
                            }
                          }
                        >
                          -
                        </span>
                      </EuiToolTip>
                    }
                    titleColor="primary"
                  >
                    <div
                      className="euiStat euiStat--centerAligned"
                    >
                      <EuiText
                        className="euiStat__description"
                        size="s"
                      >
                        <div
                          className="euiText euiText--small euiStat__description"
                        >
                          <p
                            aria-hidden={true}
                          >
                            Total agents
                          </p>
                        </div>
                      </EuiText>
                      <EuiTitle
                        className="euiStat__title euiStat__title--primary"
                        size="l"
                      >
                        <p
                          aria-hidden={true}
                          className="euiTitle euiTitle--large euiStat__title euiStat__title--primary"
                        >
                          <EuiToolTip
                            content="Go to total agents"
                            delay="regular"
                            position="top"
                          >
                            <span
                              className="euiToolTipAnchor"
                              onKeyUp={[Function]}
                              onMouseOut={[Function]}
                              onMouseOver={[Function]}
                            >
                              <span
                                className="statWithLink"
                                onBlur={[Function]}
                                onClick={[Function]}
                                onFocus={[Function]}
                                style={
                                  Object {
                                    "cursor": "pointer",
                                  }
                                }
                              >
                                -
                              </span>
                            </span>
                          </EuiToolTip>
                        </p>
                      </EuiTitle>
                    </div>
                  </EuiStat>
                </div>
              </EuiFlexItem>
              <EuiFlexItem
                key="agent-status-active"
              >
                <div
                  className="euiFlexItem"
                >
                  <EuiStat
                    description="Active agents"
                    textAlign="center"
                    title={
                      <EuiToolTip
                        content="Go to active agents"
                        delay="regular"
                        position="top"
                      >
                        <span
                          className="statWithLink"
                          onClick={[Function]}
                          style={
                            Object {
                              "cursor": "pointer",
                            }
                          }
                        >
                          -
                        </span>
                      </EuiToolTip>
                    }
                    titleColor="#007871"
                  >
                    <div
                      className="euiStat euiStat--centerAligned"
                    >
                      <EuiText
                        className="euiStat__description"
                        size="s"
                      >
                        <div
                          className="euiText euiText--small euiStat__description"
                        >
                          <p
                            aria-hidden={true}
                          >
                            Active agents
                          </p>
                        </div>
                      </EuiText>
                      <EuiTitle
                        className="euiStat__title"
                        size="l"
                      >
                        <p
                          aria-hidden={true}
                          className="euiTitle euiTitle--large euiStat__title"
                          style={
                            Object {
                              "color": "#007871",
                            }
                          }
                        >
                          <EuiToolTip
                            content="Go to active agents"
                            delay="regular"
                            position="top"
                          >
                            <span
                              className="euiToolTipAnchor"
                              onKeyUp={[Function]}
                              onMouseOut={[Function]}
                              onMouseOver={[Function]}
                            >
                              <span
                                className="statWithLink"
                                onBlur={[Function]}
                                onClick={[Function]}
                                onFocus={[Function]}
                                style={
                                  Object {
                                    "cursor": "pointer",
                                  }
                                }
                              >
                                -
                              </span>
                            </span>
                          </EuiToolTip>
                        </p>
                      </EuiTitle>
                    </div>
                  </EuiStat>
                </div>
              </EuiFlexItem>
              <EuiFlexItem
                key="agent-status-disconnected"
              >
                <div
                  className="euiFlexItem"
                >
                  <EuiStat
                    description="Disconnected agents"
                    textAlign="center"
                    title={
                      <EuiToolTip
                        content="Go to disconnected agents"
                        delay="regular"
                        position="top"
                      >
                        <span
                          className="statWithLink"
                          onClick={[Function]}
                          style={
                            Object {
                              "cursor": "pointer",
                            }
                          }
                        >
                          -
                        </span>
                      </EuiToolTip>
                    }
                    titleColor="#BD271E"
                  >
                    <div
                      className="euiStat euiStat--centerAligned"
                    >
                      <EuiText
                        className="euiStat__description"
                        size="s"
                      >
                        <div
                          className="euiText euiText--small euiStat__description"
                        >
                          <p
                            aria-hidden={true}
                          >
                            Disconnected agents
                          </p>
                        </div>
                      </EuiText>
                      <EuiTitle
                        className="euiStat__title"
                        size="l"
                      >
                        <p
                          aria-hidden={true}
                          className="euiTitle euiTitle--large euiStat__title"
                          style={
                            Object {
                              "color": "#BD271E",
                            }
                          }
                        >
                          <EuiToolTip
                            content="Go to disconnected agents"
                            delay="regular"
                            position="top"
                          >
                            <span
                              className="euiToolTipAnchor"
                              onKeyUp={[Function]}
                              onMouseOut={[Function]}
                              onMouseOver={[Function]}
                            >
                              <span
                                className="statWithLink"
                                onBlur={[Function]}
                                onClick={[Function]}
                                onFocus={[Function]}
                                style={
                                  Object {
                                    "cursor": "pointer",
                                  }
                                }
                              >
                                -
                              </span>
                            </span>
                          </EuiToolTip>
                        </p>
                      </EuiTitle>
                    </div>
                  </EuiStat>
                </div>
              </EuiFlexItem>
              <EuiFlexItem
                key="agent-status-pending"
              >
                <div
                  className="euiFlexItem"
                >
                  <EuiStat
                    description="Pending agents"
                    textAlign="center"
                    title={
                      <EuiToolTip
                        content="Go to pending agents"
                        delay="regular"
                        position="top"
                      >
                        <span
                          className="statWithLink"
                          onClick={[Function]}
                          style={
                            Object {
                              "cursor": "pointer",
                            }
                          }
                        >
                          -
                        </span>
                      </EuiToolTip>
                    }
                    titleColor="#FEC514"
                  >
                    <div
                      className="euiStat euiStat--centerAligned"
                    >
                      <EuiText
                        className="euiStat__description"
                        size="s"
                      >
                        <div
                          className="euiText euiText--small euiStat__description"
                        >
                          <p
                            aria-hidden={true}
                          >
                            Pending agents
                          </p>
                        </div>
                      </EuiText>
                      <EuiTitle
                        className="euiStat__title"
                        size="l"
                      >
                        <p
                          aria-hidden={true}
                          className="euiTitle euiTitle--large euiStat__title"
                          style={
                            Object {
                              "color": "#FEC514",
                            }
                          }
                        >
                          <EuiToolTip
                            content="Go to pending agents"
                            delay="regular"
                            position="top"
                          >
                            <span
                              className="euiToolTipAnchor"
                              onKeyUp={[Function]}
                              onMouseOut={[Function]}
                              onMouseOver={[Function]}
                            >
                              <span
                                className="statWithLink"
                                onBlur={[Function]}
                                onClick={[Function]}
                                onFocus={[Function]}
                                style={
                                  Object {
                                    "cursor": "pointer",
                                  }
                                }
                              >
                                -
                              </span>
                            </span>
                          </EuiToolTip>
                        </p>
                      </EuiTitle>
                    </div>
                  </EuiStat>
                </div>
              </EuiFlexItem>
              <EuiFlexItem
                key="agent-status-never_connected"
              >
                <div
                  className="euiFlexItem"
                >
                  <EuiStat
                    description="Never connected agents"
                    textAlign="center"
                    title={
                      <EuiToolTip
                        content="Go to never connected agents"
                        delay="regular"
                        position="top"
                      >
                        <span
                          className="statWithLink"
                          onClick={[Function]}
                          style={
                            Object {
                              "cursor": "pointer",
                            }
                          }
                        >
                          -
                        </span>
                      </EuiToolTip>
                    }
                    titleColor="#646A77"
                  >
                    <div
                      className="euiStat euiStat--centerAligned"
                    >
                      <EuiText
                        className="euiStat__description"
                        size="s"
                      >
                        <div
                          className="euiText euiText--small euiStat__description"
                        >
                          <p
                            aria-hidden={true}
                          >
                            Never connected agents
                          </p>
                        </div>
                      </EuiText>
                      <EuiTitle
                        className="euiStat__title"
                        size="l"
                      >
                        <p
                          aria-hidden={true}
                          className="euiTitle euiTitle--large euiStat__title"
                          style={
                            Object {
                              "color": "#646A77",
                            }
                          }
                        >
                          <EuiToolTip
                            content="Go to never connected agents"
                            delay="regular"
                            position="top"
                          >
                            <span
                              className="euiToolTipAnchor"
                              onKeyUp={[Function]}
                              onMouseOut={[Function]}
                              onMouseOver={[Function]}
                            >
                              <span
                                className="statWithLink"
                                onBlur={[Function]}
                                onClick={[Function]}
                                onFocus={[Function]}
                                style={
                                  Object {
                                    "cursor": "pointer",
                                  }
                                }
                              >
                                -
                              </span>
                            </span>
                          </EuiToolTip>
                        </p>
                      </EuiTitle>
                    </div>
                  </EuiStat>
                </div>
              </EuiFlexItem>
              <EuiFlexItem>
                <div
                  className="euiFlexItem"
                />
              </EuiFlexItem>
            </div>
          </EuiFlexGroup>
        </div>
      </EuiPage>
    </withErrorBoundary(Stats)>
  </ErrorBoundary>
</Component>
`;
